### ISSUE TITLE
[Accessibility] Command history is not readable in terminal input field via Up/DownArrow

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#172149]
CC @meganrogge

Tested with NVDA and JAWS on Windows 11.

## Reproducible Steps

1. Load screen reader.

1. Launch terminal.

1. Type `echo hello` (hit enter).

1. Type `echo world` (hit enter).

1. In terminal input field, press UpArrow and DownArrow to navigate command history and see if you can hear them.

* Notes: Nothing is heard. Screen reader keeps saying "blank."

## Expected Behavior

Command history needs to be accessible and readable to screen reader.

## System Info

Version: 1.75.0-insider (system setup)
Commit: 0a6c277e4faa44a6b6eb82ddca8bfeccf7fecef6
Date: 2023-01-24T05:22:55.489Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

**Comment by meganrogge** [comment]
@jooyoungseo do you know if this has ever worked in the VS Code terminal IE is a regression? Just tried in version 1.72 and doesn't work there.

**Comment by meganrogge** [comment]
As a not ideal workaround in the meantime, you can use the `Run Recent Command` command to go through what you've just run. The default keyboard shortcut is `ctrl+r` when the terminal is focused

**Comment by jooyoungseo** [comment]
@meganrogge

This is not a regression, but just a long-standing unresolved issue.

Thanks for the stopgap workaround.

Tested on Windows 11:

`Terminal: Run Recent Command...` does not work in terminal input area

**Comment by meganrogge** [comment]
`Terminal: Run recent command...` should work, do you have `terminal.integrated.shellIntegration.enabled`?

**Comment by jooyoungseo** [comment]
@meganrogge  Yes, that's already enabled by default. Does not work even via the manual execution through command palette.

**Comment by meganrogge** [comment]
@jooyoungseo sorry if this is obvious to you already, but you have to press the down arrow immediately after running that to go to the most recently run command. then pressing the down arrow again will take you to the second most recently one run. to run it, you'd press enter

**Comment by jooyoungseo** [comment]
@meganrogge It's so odd. It's not working at all on my end. It keeps saying "blank." Tested with NVDA and JAWS on Windows 11

Version: 1.75.0-insider (system setup)
Commit: 0a6c277e4faa44a6b6eb82ddca8bfeccf7fecef6
Date: 2023-01-24T05:22:55.489Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

**Comment by meganrogge** [comment]
Just tried on Windows, and I see what you mean there @jooyounseo. I'm only seeing commands run in previous sessions - ones from my current session aren't getting stored there. We'll look into a fix for this ASAP, as it was working recently

**Comment by meganrogge** [comment]
commented in that issue, but shell integration relies upon `PSReadline`, which unfortunately is disabled when a screen reader is detected. that is why `Run Recent Command` does not work

**Comment by jooyoungseo** [comment]
@meganrogge -- Aw, that makes sense. Thanks for investigating this issue.

**Comment by meganrogge** [comment]
Np, will keep this on my radar for upcoming iterations

**Comment by meganrogge** [comment]
The workaround IE using `Run Recent Command` should work in insider's tomorrow as a result of getting shell integration to work when screen readers are used on windows 3912d68fdf20b25c003600513e64f0f20210e5e3

**Comment by steverep** [comment]
I agree this is a longstanding and very prohibitive issue.  Note there also is multiple duplicate issues:
- https://github.com/microsoft/vscode/issues/185163
- https://github.com/microsoft/vscode/issues/186904

**Comment by Tyriar** [comment]
Let's track in https://github.com/microsoft/vscode/issues/186904

/duplicate
